packages/ui-components/src/Button/Button.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FunctionComponent } from "react";
+import classnames from "classnames/bind";
+
+import objectToClassnames from "../utils/objectToClassnames";
+
+import styles from "./Button.module.scss";
+
+type ButtonIntent =
+  | "primary-success"
+  | "primary-danger"
+  | "secondary"
+  | "tertiary";
+type ButtonType = "button" | "submit" | "reset";
+type ButtonSize = "standard" | "small";
+
+interface ButtonProps {
+  id?: string;
+  ariaLabel?: string;
+  ariaLabelledBy?: string;
+  intent?: ButtonIntent;
+  type?: ButtonType;
+  size?: ButtonSize;
+  onClick?: () => void;
+}
+
+const cx = classnames.bind(styles);
+
+export const Button: FunctionComponent<ButtonProps> = ({
+  intent = "secondary",
+  type = "button",
+  size = "standard",
+  children,
+  ...props
+}) => {
+  const classNames = cx("button", objectToClassnames({ intent, size }));
+
+  if (children !== undefined) {
+    return (
+      <button className={classNames} type={type} {...props}>
```

Comment:
Based on my comment above, here we would keep the _rest spread_ of props, but now those props are properly typed. And we could drop the `type` prop to simple be

```jsx
<button className={className} {...props}>
```